packages/components/src/templates/next/components/complex/ContactInformation/components/HomepageContactInformationUI.tsx

```diff
@@ -0,0 +1,199 @@
+import DOMPurify from "isomorphic-dompurify"
+
+import type { ContactInformationUIProps } from "~/interfaces"
+import { tv } from "~/lib/tv"
+import { LinkButton } from "../../../internal/LinkButton"
+import {
+  commonContactInformationStyles,
+  commonContactMethodStyles,
+} from "./common"
+import { ContactMethod, LoadingContactMethod } from "./ContactMethod"
+
+const createHomepageContactInformationStyles = tv({
+  extend: commonContactInformationStyles,
+  slots: {
+    ...commonContactInformationStyles.slots,
+    container: "gap-12 py-12 md:py-16",
+    contactMethodsContainer: "grid grid-cols-1 gap-10",
+  },
+  variants: {
+    numberOfContactMethods: {
+      2: {
+        container: "md:flex-row md:gap-10",
+        titleAndDescriptionContainer: "max-w-[24.5rem]",
+        contactMethodsContainer:
+          "md:flex md:flex-1 md:flex-col lg:grid lg:grid-cols-2",
+      },
+      3: {
+        titleAndDescriptionContainer: "lg:max-w-3xl",
+        contactMethodsContainer: "md:grid md:grid-cols-3",
+      },
+    },
+    isBottomButton: {
+      true: {},
+      false: {},
+    },
+  },
+  compoundVariants: [
+    {
+      numberOfContactMethods: 2,
+      isBottomButton: false,
+      class: {
+        urlButtonContainer: "mt-10 hidden md:mx-0 md:block",
+      },
+    },
+    {
+      numberOfContactMethods: 2,
+      isBottomButton: true,
+      class: {
+        urlButtonContainer: "block md:mx-0 md:hidden",
+      },
+    },
+    {
+      numberOfContactMethods: 3,
+      isBottomButton: false,
+      class: {
+        urlButtonContainer: "hidden",
+      },
+    },
+    {
+      numberOfContactMethods: 3,
+      isBottomButton: true,
+      class: {
+        urlButtonContainer: "block",
+      },
+    },
+  ],
+})
+
+const createHomepageContactMethodStyles = tv({
+  extend: commonContactMethodStyles,
+  variants: {
+    numberOfContactMethods: {
+      2: {},
+      3: {
+        container: "md:items-center",
+        textContainer: "md:items-center",
+        label: "md:text-center",
+        valuesAndCaptionContainer: "md:items-center",
+        value: "md:text-center",
+      },
+    },
+  },
+})
+
+const MAX_CONTACT_METHODS_FOR_HOMEPAGE = 3
+
+export const HomepageContactInformationUI = ({
+  whitelistedMethods,
+  title,
+  description,
+  methods,
+  referenceLinkHref,
+  label,
+  LinkComponent,
+  isLoading,
+  acceptHtmlTags = false,
+}: ContactInformationUIProps) => {
+  const filteredMethods = whitelistedMethods
+    ? methods.filter(
+        (method) => method.method && whitelistedMethods.includes(method.method),
+      )
+    : methods
+
+  const numberOfContactMethods = isLoading
+    ? MAX_CONTACT_METHODS_FOR_HOMEPAGE
+    : filteredMethods.length >= MAX_CONTACT_METHODS_FOR_HOMEPAGE
+      ? MAX_CONTACT_METHODS_FOR_HOMEPAGE
+      : 2
```

Comment:
nit: I think this works only when `MAX_CONTACT_METHODS_FOR_HOMEPAGE` is 3 like it currently is, but gets weird when it is increased to say 4, cos we don't have the styles for it so more than just the const variable change would be required. There is some inherent link between this variable and the variants available in `createHomepageContactInformationStyles` and `createHomepageContactMethodStyles`.

Is it possible for us to enhance the guarantee that the result of `numberOfContactMethods` is either 2 or 3, regardless of the setting of `MAX_CONTACT_METHODS_FOR_HOMEPAGE` or the result of `filteredMethods.length`?

Reply:
udpated https://github.com/opengovsg/isomer/pull/1534/commits/165d29611467aaeb6cc98d890d224c94cfa34a77